pgxn/neon/control_plane_connector.c

```diff
@@ -454,16 +454,15 @@ MergeTable()
 		}
 		hash_destroy(old_table->role_table);
 	}
+	pfree(old_table);
 }
 
 static void
 PopTable()
 {
-	/*
-	 * Current table gets freed because it is allocated in aborted
-	 * subtransaction's memory context.
-	 */
+	DdlHashTable *old_table = CurrentDdlTable;
 	CurrentDdlTable = CurrentDdlTable->prev_table;
+	pfree(old_table);
```

Comment:
Does this leak `old_table->db_table` and `old_table->role_table` ?

Reply:
This hashes are created in the current transaction context.
So they are deallocated at the end of transaction. No explicit reallocation iso needed (as explained in the removed comment).

I thought about lazy allocation of `DdlHashTable`. But it seems to be untriovial: we in any case need some stack to represent hierarchy of transactions.

So allocation of `DdlHashTable` in top transaction context seems to be the minimal possible fix.